src/main/java/org/jitsi/videobridge/xmpp/MediaStreamTrackFactory.java

```diff
@@ -29,43 +31,150 @@
+     */
+    private static final String ENABLE_SVC_PNAME = "org.jitsi" +
+        ".videobridge.ENABLE_SVC";
+
+    /**
+     * The default number of temporal layers to use for VP8 simulcast.
+     *
+     * FIXME: hardcoded ugh.. this should be either signaled or somehow included
+     * in the RTP stream.
+     */
+    private static final int VP8_SIMULCAST_TEMPORAL_LAYERS = 3;
+
+    /**
+     * The default number of spatial layers to use for VP9 SVC.
+     *
+     * FIXME: hardcoded ugh.. this should be either signaled or somehow included
+     * in the RTP stream.
+     */
+    private static final int VP9_SVC_SPATIAL_LAYERS = 3;
+
+    /**
+     * The default number of spatial layers to use for VP9 SVC.
+     *
+     * FIXME: hardcoded ugh.. this should be either signaled or somehow included
+     * in the RTP stream.
+     */
+    private static final int VP9_SVC_TEMPORAL_LAYERS = 3;
+
+    /**
+     * A boolean that's controlling whether or not to enable SVC filtering for
+     * scalable video codecs.
+     */
+    private static final boolean ENABLE_SVC
+        = cfg.getBoolean(ENABLE_SVC_PNAME, false);
+
+    /**
+     * Creates encodings.
      *
      * @param track the track that will own the temporal encodings.
      * @param primary the array of the primary SSRCs for the simulcast streams.
      * @param rtx the array of the RTX SSRCs for the simulcast streams.
-     * @param tempoLen the number of temporal encodings per simulcast stream.
+     * @param spatialLen the number of spatial encodings per simulcast stream.
+     * @param temporalLen the number of temporal encodings per simulcast stream.
      * @return an array that holds the simulcast encodings.
      */
-    public static RTPEncodingDesc[] createRTPEncodings(
-        MediaStreamTrackDesc track, long[] primary, long[] rtx, int tempoLen)
+    private static RTPEncodingDesc[] createRTPEncodings(
+        MediaStreamTrackDesc track, long[] primary, long[] rtx,
+        int spatialLen, int temporalLen)
     {
         RTPEncodingDesc[] rtpEncodings
-            = new RTPEncodingDesc[primary.length * tempoLen];
-
-        for (int i = 0; i < rtpEncodings.length; i++)
+            = new RTPEncodingDesc[primary.length * spatialLen * temporalLen];
+
+        // this loop builds a subjective quality index array that looks like
+        // this:
+        //
+        // [s0t0, s0t1, s0t2, s1t0, s1t1, s1t2, s2t0, s2t1, s2t2]
+        //
+        // The spatial layer is offered either by simulcast (VP8) or spatial
+        // scalability (VP9). Exotic cases might do simulcast + spatial
+        // scalability.
+
+        if (spatialLen < 2 /* simulcast */)
         {
-            int streamIdx = i / tempoLen, tempoIdx = i % tempoLen;
+            // this O(n) loop is more efficient than the O(n^3) loop below, and
+            // it works well for the simulcast case, so we keep it.
+            for (int i = 0; i < rtpEncodings.length; i++)
```

Comment:
If the only reason for this code is optimization I would rather remove it (less logic for humans to understand, less space for a bug to hide). It seems like we expect n to be on the order of 9, so n^3 shouldn't be a problem.